packages/components/src/components/Pagination/usePagination.tsx

```diff
@@ -0,0 +1,64 @@
+import { useMemo } from "react";
+
+const DOTS = "...";
+
+const range = (start: number, end: number): string[] => {
+  const length = end - start + 1;
+  return Array.from({ length }, (_, idx) => `${idx + start}`);
+};
+
+export const usePagination = (
+  totalCount: number,
+  siblingCount = 1,
+  currentPage: number
+): (number | string)[] => {
+  return useMemo(() => {
+    // Pages count is determined as siblingCount + firstPage + lastPage + currentPage + 2*DOTS
```

Comment:
This comment should be reflected in the code so that we don't need the comment anymore.